common/oatbox/filesystem/FileSystemService.php

```diff
@@ -23,33 +23,58 @@
 use League\Flysystem\AdapterInterface;
 use common_exception_Error;
 use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use \League\Flysystem\Filesystem as FlyFileSystem;
+use League\Flysystem\FilesystemInterface;
  /**
  * A service to reference and retrieve filesystems
  */
 class FileSystemService extends ConfigurableService
 {
     const SERVICE_ID = 'generis/filesystem';
-    
+
     const OPTION_FILE_PATH = 'filesPath';
-    
+
     const OPTION_ADAPTERS = 'adapters';
-    
+
+    const OPTION_DIRECTORIES = 'dirs';
+
     const FLYSYSTEM_ADAPTER_NS = '\\League\\Flysystem\\Adapter\\';
-    
+
     const FLYSYSTEM_LOCAL_ADAPTER = 'Local';
-    
+
     private $filesystems = array();
-    
+
     /**
      * 
      * @param $id
      * @return \oat\oatbox\filesystem\Directory
      */
     public function getDirectory($id)
     {
-        $directory = new Directory($id, '');
-        $directory->setServiceLocator($this->getServiceLocator());
-        return $directory;
+        return $this->propagate(new Directory($id, ''));
```

Comment:
All this and following functions are a directory repository. Would you consider making it a separate object to lighten this class?